src/cdomains/intDomain.ml

```diff
@@ -492,9 +492,13 @@ struct
     | Some (a, b) ->
       if a = b && b = i then `Eq else if Ints_t.compare a i <= 0 && Ints_t.compare i b <=0 then `Top else `Neq
 
+  let set_overflow_flag ik =
+    if Cil.isSigned ik && !GU.in_verifying_stage then
+      Goblintutil.did_overflow := true
+
   let norm ik = function None -> None | Some (x,y) ->
-    if Ints_t.compare x y > 0 then None
-    else if Ints_t.compare (min_int ik) x > 0 || Ints_t.compare (max_int ik) y < 0 then top_of ik
+    if Ints_t.compare x y > 0 then (set_overflow_flag ik; None)
```

Comment:
Do we need to set the flag here in fact?

Reply:
Not setting it here causes us to produce correct results for more cases, but we also the become unsound for 3 cases. I'll see what's going on there.

Reply:
Setting it here is not necessary, but it needs to be set in `shift` operations when we don't bother to find out the actual result and simply go to top.
This means we answer correctly on one fewer task, but is the right thing to do.